extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -640,7 +640,8 @@ export class AmpIframe extends AMP.BaseElement {
       return;
     }
     this.isDisplayed_ = isDisplayed;
-    if (!isDisplayed && this.iframe_) {
+    const hasOwner = !!this.element.getOwner();
```

Comment:
I think this check should be before setting `this.isDisplayed_`, so that if we become unowned we can still unload.

Reply:
Hmm. Probably even safer to remove the bailout in case of unchanged isDisplay

Reply:
Even that change though would still not really address the issue that the fn call needs to be trigger when ownership changes as opposed to when display status happens to change after an ownership change. 🤔 . Probably worth fixing separately